File: M2/PARV/TP Bonhomme de neige/src/Sujet.cpp
#include "Sujet.h"


Sujet::Sujet()
{
}


Sujet::~Sujet()
{
	listeObservateur.clear();
}


void Sujet::Notifie(){
	for(int i=0;i<listeObservateur.size();i++){
		listeObservateur[i]->MiseAJour();
	}
}

void Sujet::Abonne(IObservateur *o){
	listeObservateur.push_back(o);
}

File: Algorithme Jointure BDD/src/Interface/Sujet.java
package Interface;

import java.util.ArrayList;
import java.util.List;

public class Sujet {
	
	private List<Observateur> observateurs;
	
	public Sujet(){
		observateurs = new ArrayList<Observateur>();
	}
	
	public void Abonne(Observateur o){
		observateurs.add(o);
	}
	
	public void Notifie(){
		for(Observateur o : observateurs){
			o.MiseAJour(this);
		}
	}
	
}
